File: AppointmentSchedulerAPI/layers/BusinessLogicLayer/Model/SystemNotification.cs
using AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model.Types.Notification;

namespace AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model
{
    public class SystemNotification : NotificationBase
    {
        public SystemNotification()
        {
            this.Type = NotificationType.SYSTEM_NOTIFICATION;
        }

        public required SystemNotificationCodeType? Code { get; set; }
        public required SystemNotificationSeverityCodeType? Severity { get; set;}
        
        
    }
}

File: AppointmentSchedulerWeb/src/view-model/business-entities/notification/system-notification.ts
import { NotificationType } from "../types/notification.types";
import { SystemNotificationCodeType } from "../types/system-notification-code.types";
import { SystemNotificationSeverityCodeType } from "../types/system-notification-severity-code.types";
import { NotificationBase } from "./notification-base";


export class SystemNotification extends NotificationBase {
  constructor(
    createdAt: Date,
    uuid: string,
    message: string,
    type: NotificationType,
    public code: SystemNotificationCodeType,
    public severity: SystemNotificationSeverityCodeType,
  ) {
    super(createdAt, uuid, message, type);
  }

}
